mne/time_frequency/tfr.py

```diff
@@ -1137,156 +1434,151 @@ def save(self, fname, overwrite=False):
+    coherence.
 
     Parameters
     ----------
-    fname : string
-        The file name, which should end with -tfr.h5
-    tfr : AverageTFR instance, or list of AverageTFR instances
-        The TFR dataset, or list of TFR datasets, to save in one file.
-        Note. If .comment is not None, a name will be generated on the fly,
-        based on the order in which the TFR objects are passed
-    overwrite : bool
-        If True, overwrite file (if it exists). Defaults to False.
+    info : Info
+        The measurement info.
+    data : ndarray, shape (n_epochs, n_channels, n_freqs, n_times)
+        The data.
+    times : ndarray, shape (n_times,)
+        The time values in seconds.
+    freqs : ndarray, shape (n_freqs,)
+        The frequencies in Hz.
+    comment : str | None, defaults to None
+        Comment on the data, e.g., the experimental condition.
+    method : str | None, defaults to None
+        Comment on the method used to compute the data, e.g., morlet wavelet.
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see mne.verbose).
 
-    See Also
-    --------
-    read_tfrs
+    Attributes
+    ----------
+    ch_names : list
+        The names of the channels.
 
     Notes
     -----
-    .. versionadded:: 0.9.0
+    .. versionadded:: 0.12.0
     """
-    out = []
-    if not isinstance(tfr, (list, tuple)):
-        tfr = [tfr]
-    for ii, tfr_ in enumerate(tfr):
-        comment = ii if tfr_.comment is None else tfr_.comment
-        out.append(_prepare_write_tfr(tfr_, condition=comment))
-    write_hdf5(fname, out, overwrite=overwrite, title='mnepython')
+    @verbose
+    def __init__(self, info, data, times, freqs, comment=None,
+                 method=None, verbose=None):
+        self.info = info
+        if data.ndim != 4:
+            raise ValueError('data should be 4d. Got %d.' % data.ndim)
+        n_epochs, n_channels, n_freqs, n_times = data.shape
+        if n_channels != len(info['chs']):
+            raise ValueError("Number of channels and data size don't match"
+                             " (%d != %d)." % (n_channels, len(info['chs'])))
+        if n_freqs != len(freqs):
+            raise ValueError("Number of frequencies and data size don't match"
+                             " (%d != %d)." % (n_freqs, len(freqs)))
+        if n_times != len(times):
+            raise ValueError("Number of times and data size don't match"
+                             " (%d != %d)." % (n_times, len(times)))
+        self.data = data
+        self.times = np.array(times)
+        self.freqs = np.array(freqs)
+        self.comment = comment
+        self.method = method
 
+    def __repr__(self):
+        s = "time : [%f, %f]" % (self.times[0], self.times[-1])
+        s += ", freq : [%f, %f]" % (self.freqs[0], self.freqs[-1])
+        s += ", epochs : %d" % self.data.shape[0]
+        s += ', channels : %d' % self.data.shape[1]
+        return "<EpochsTFR  |  %s>" % s
```

Comment:
I think it'd be useful to show a size estimate as well, like @Eric89GXL did [here](https://github.com/mne-tools/mne-python/blob/master/mne/io/base.py#L1960). Memory usage will probably be a particular concern w/ etfrs